preprocess/docenc_main.cc

```diff
@@ -0,0 +1,258 @@
+#include <algorithm>
+#include <cstdio>
+#include <cstdlib>
+#include <iostream>
+#include <sstream>
+#include <stdexcept>
+#include <unistd.h>
+#include <vector>
+#include "util/file_stream.hh"
+#include "util/file_piece.hh"
+#include "util/string_piece.hh"
+#include "preprocess/base64.hh"
+
+using namespace std;
+
+enum Mode {
+	COMPRESS,
+	DECOMPRESS
+};
+
+string prefix_lines(string const &input, string const prefix)
+{
+	stringstream sin(input);
```

Comment:
stringstream is crazy slow. Seems you just want to split on newlines so use `util::TokenIter<util::SingleCharacter, false>(input, '\n')`